benchmarks/bench_ml.py

```diff
@@ -0,0 +1,451 @@
+    datasets={
+        "Boston":{
+            OUTPUT_TYPE: FMT_NUMBER,
+            DATASET_LOADER: lambda : load_data_target(skd.load_boston())
+        },
+        "Housing":{
+            OUTPUT_TYPE: FMT_NUMBER,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_california_housing())
+        },
+        #
+        #   classification datasets go here
+        #
+        "Leukemia":{
+            OUTPUT_TYPE: FMT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('leukemia'))
+        },
+        "Climate Model Crashes":{
+            OUTPUT_TYPE: FMT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('climate-model-simulation-crashes', target_name=-1))
+        },
+    }
+
+    @staticmethod
+    def supports(model, dataset):
+        """
+        Used to determine if model can be trained on particular dataset.
+        :param model: string, name of model class
+        :param dataset: string, name of dataset
+        :return: bool
+        """
+
+        if not dataset in TesterClass.datasets:
+            return False, "unknown dataset"
+
+        dst = TesterClass.datasets[dataset]
+
+        if not model in TesterClass.models:
+            return False, "unknown model"
+
+        mod = TesterClass.models[model]
+
+        if not dst[OUTPUT_TYPE] == mod[OUTPUT_TYPE]:
+            return False, "model cannot be applied to the dataset"
+
+        return True, None
+
+    def __init__(self, model, dataset):
+        supports, reason_if_not = TesterClass.supports(model, dataset)
+
+        if not supports:
+            raise BaseException(reason_if_not)
+
+        X, Y = self.datasets[dataset][DATASET_LOADER]()
+
+        self.dataset = split_normalize(X, Y)
+        self.model_class = TesterClass.models[model]
+
+    def evaluate(self, point):
+        """
+        Evaluates model on loaded dataset for particular setting of hyperparameters.
+        :param point: dict, pairs of parameter names and corresponding values
+        :return: score (more is better!) for some specific point
+        """
+
+        X, Y, Xv, Yv = self.dataset
+        cls = self.model_class[MODEL_BACKEND]
+
+        # apply transformation to model parameters, for example exp transformation
+        point_mapped = {}
+
+        for param in point.keys():
+            v = point[param]
+            v = self.model_class[MODEL_PARAMETERS][param][PARAM_MAP](v)
+            point_mapped[param] = v
+
+        model = cls(**point_mapped)
+
+        try:
+            model.fit(X, Y)
+            r = model.score(Xv, Yv)
+        except BaseException as ex:
+            r = 0.0 # on error: return assumed smallest value of objective function
+
+        # while negative values could be informative, they could be very large also,
+        # which could mess up the optimization procedure. Suggestions are welcome.
+        r = max(r, 0.0)
```

Comment:
I considered using MSE, but this is why I did not. The larger the values of outputs, the larger the  magnitude of the values of the MSE. So it might happen that surrogate A is good for one specific task with large values of output (and thus large MSE), which makes a large contribution to  average performance over multiple tasks, compared to surrogate B which is better for other tasks, but other tasks do not make a large contribution due to values of output being relatively small. Thus it might seem that surrogate A performs on average better than B, even though if you look at case by case performance it is not. One could solve this by normalizing values of MSE, but this is what r2_score already does. Additionally, range of values of  r2_score is in most cases in range of [0..1.0], which is also the range for accuracy.